src/interpreter/executor/interpret.js

```diff
@@ -0,0 +1,332 @@
+            return;
+        }
+        
+        currentDataHold.name = buffer[0];
+        currentDataHold.value = buffer[1];
+
+    } else {
+        // We do not need to parse buffer since we already did before falling
+        // falling into this method, and we know there is no '=' in the buffer
+        currentDataHold.name = buffer;
+    }
+
+    currentLineRep.push(currentDataHold);
+}
+
+
+// Boolean function to determine if we should treat default state as variable
+// May not need this if there are not many cases
+function shouldBeVariable(buffer) {
+
+    var isVar = true;
+
+    if (!isNaN(parseInt(buffer))) {
+        isVar = false;
+    }
+
+    return isVar;
+}
+
+// Band Aid for things not yet covered
+function isNotCovered(buffer) {
+
+    var isNotCovered = false;
+
+    if (buffer.includes("function")) {
+        isNotCovered = true;
+    } else if (buffer.includes("console")) {
+        isNotCovered = true;
+    } else if (buffer.includes("{")) {
+        isNotCovered = true;
+    } else if (buffer.includes("}")) {
+        isNotCovered = true;
+    } else if (buffer.includes("()")) {
+        isNotCovered = true;
+    }
+
+    return isNotCovered;
+}
+
+// Handle our current state cases
+function evalState(buffer) {
+
+    switch (currentState) {
+        
+        //We should not ever hit this case but let's put it in here just in case
+        case stateEnum.DEFAULT:
+            findBufferState(buffer);
+            break;
+
+        case stateEnum.ACCEPTING_VAR:
+            buffer = acceptingVar(buffer);
+            break;
+
+        case stateEnum.ASSIGNING_VAR:
+            break;
+    }
+    return buffer;
+}
+
+// If we are in accepting_var state
+function acceptingVar(buffer) {
+
+    if (buffer[buffer.length - 1] == '=') {
+        buffer = buffer.substring(0, buffer.length - 1);
+        
+        currentDataHold.name = buffer
+
+        //Reset Variable to start accepting value
+        buffer = "";
+        currentState = stateEnum.ASSIGNING_VAR;
+    }
+    return buffer;
+}
+
+//Evaluate our buffer to try to find a state
+function findBufferState(buffer) {
+
+    switch (buffer) {
+        case "var":
+            currentState = stateEnum.ACCEPTING_VAR;
+            currentDataHold = new Variable();
+            currentLineRep.push(currentDataHold);
+            return "";
+            break;
+        default:
+            currentState = stateEnum.DEFAULT;
+            break;
+    }
+
+    return buffer;
+}
+
+// Global evaluate function called on the step and run click
+function evaluate(inputCode) {
+    
+    if (!setup(inputCode)) {
+        return false;
+    }    
+
+    //While we haven't evaluated all of our code
+    while (currentStep < userCode.length - 1) {
+        interpretLine(currentStep);
+        currentStep += 1;
+    }
+
+    masterRepToString(masterRep);
+    exitInterpret();
+    return masterRep;
+}
+
+// Print out our representation
+function masterRepToString(representation) {
```

Comment:
Is this just to get up and running? Or do we plan to keep this around?

Reply:
Mostly for my testing I can take it out for the PR if you would like.  However, we can use this function to for travis testing possibly.

Reply:
It's fine there for now. My only issue for testing is that when I call `evaluate` i end up with empty array, but all the right data got logged to console. Will discuss w/ you on slack.